tests/e2e/executors/base.py

```diff
@@ -20,3 +21,9 @@ def prepare_then_cleanup(self):
 
     def prepare(self):
         raise NotImplementedError
+
+    @classmethod
+    def prepare_global(cls):
+        if cls._global_resource_tmp_dir is None:
+            with utils.tmp_dir() as tmp_dirpath:
+                cls._global_resource_tmp_dir = tmp_dirpath
```

Comment:
The directory will be cleaned up after you leave the scope of `with` which is not exactly what you want, do you? You probably want to clean it up once executer has done its job. Otherwise I don't see any logic associated with purging this directory anywhere else. 

I also don't see much value having the `prepare_global` logic in the base class. Why can't this logic be a part of `prepare` of the `VisualBasicExecutor`?

Reply:
> You probably want to clean it up once executer has done its job. 

> Why can't this logic be a part of prepare of the VisualBasicExecutor?

I need to create this directory only once for the whole test session. It allows to decrease testing time for ~5 minutes. `prepare` is called for every executor which is not the thing I need. Please share more appropriate solution to achieve this goal.

Let me show what I need by example:
```diff
diff --git a/.travis.yml b/.travis.yml
index e5e2203..51e50e3 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -30,4 +30,4 @@ script:
   # Then we install it as a library, remove source code and run e2e tests.
   - python setup.py install
   - rm -rfd m2cgen/
-  - pytest -v tests/e2e/
+  - pytest -v --capture=no tests/e2e/
diff --git a/tests/e2e/executors/visual_basic.py b/tests/e2e/executors/visual_basic.py
index dbcaacb..f055488 100644
--- a/tests/e2e/executors/visual_basic.py
+++ b/tests/e2e/executors/visual_basic.py
@@ -51,6 +51,7 @@ class VisualBasicExecutor(base.BaseExecutor):
         self.model_ast = assembler_cls(model).assemble()

     def predict(self, X):
+        print("!!!{}!!!".format(self.target_exec_dir))
         exec_args = [os.path.join(self.target_exec_dir, self.project_name)]
         exec_args.extend(map(str, X))
         return utils.predict_from_commandline(exec_args)
```

Please notice that the directory is the same during the whole session. That is what I need to speed up tests.

Link to huge part of a huge log, because GitHub doesn't allow to paste it here in comment: https://pastebin.com/mZ65Unti.

Reply:
Can you perhaps use a module-scoped pytest fixture for this?
```
@pytest.fixture(scope="module", autouse=True)
def my_fixture():
    print ('Creating a VB executor dir')
    # Create dir
    yield vb_dir
    # Remove dir
    print ('TEAR DOWN')
```

Reply:
It was my first attempt to achieve the needed behavior. Unfortunately, I'm not familiar with `pytest` enough and didn't get it how to use a created dir inside executor code.

Reply:
I think you should put the fixture into `conftest.py` or into the `test_e2e.py` itself. Then pass it as an argument into the test_e2e test like:
```
def test_e2e(estimator, executor_cls, model_trainer, is_fast, my_fixture):
```
`my_fixture` here would be a path to a directory that you yielded in the fixture's body. From there you might want to pass the value into the executor's constructor. This way you'd have to make other constructors accept `**kwargs`. 

Were there any particular difficulties that you encountered? We can discuss them here.

Reply:
Thanks for the guide! I did it. However, actually I need to set up VB project in that temporary dir. Is it acceptable to do executor-dependent stuff inside global fixture? If you are OK with it, I'll replace class method with such fixture.

Reply:
@StrikerRUS In that fixture I only suggest to setup and cleanup the temporary directory that can be passed into the executor itself where VB-specific preparations can be made. I'd rather avoid having VB specific stuff in that fixture though.

Reply:
Ah, got it! Thanks again for the explanation! Will make changes in a few hours.